core/src/main/java/org/elasticsearch/cluster/routing/RoutingService.java

```diff
@@ -62,6 +64,10 @@
     private volatile boolean routingTableDirty = false;
 
     private volatile Future scheduledRoutingTableFuture;
```

Comment:
to be honest, I really want to get away from this scheduling completely, I started to try and do it in this change, and it became bigger and non relevant (reroute after node added in disco). My goal is to have a single place in this class to handle calling reroute, use it in allocator and shard failed one, and remove completely the scheduled nature of it, just prompt reroute. For this reason, I prefer not to rely on this time base logic, and I think its nicely encapsulated into the delay nature of when we reroute.